plugins/outputs/influxdb_v2/errors.go

```diff
@@ -0,0 +1,51 @@
+package influxdb_v2
+
+import (
+	"fmt"
+	"time"
+)
+
+type genericRespError struct {
+	Code      string
+	Message   string
+	Line      *int32
+	MaxLength *int32
+}
+
+func (g genericRespError) Error() string {
+	errString := fmt.Sprintf("%s: %s", g.Code, g.Message)
+	if g.Line != nil {
+		return fmt.Sprintf("%s - line[%d]", errString, g.Line)
+	} else if g.MaxLength != nil {
+		return fmt.Sprintf("%s - maxlen[%d]", errString, g.MaxLength)
+	}
+	return errString
+}
+
+type APIError struct {
+	Err        error
+	StatusCode int
+	Retryable  bool
+}
+
```

Comment:
Does it make sense to have a constructor for APIError?

```
func NewAPIError(err error, statusCode int, retryable bool) *APIError {
    return &APIError{
        Err:        err,
        StatusCode: statusCode,
        Retryable:  retryable,
    }
}
```

Reply:
I don't think it makes sense. For retryable errors you would have
```golang
return NewAPIError(err, statusCode, true)
```
vs
```golang
return &APIError{Err: err, StatusCode: statusCode, Retryable: true)
```

and for non-retryable errors you would even add a parameter. So you won't save anything but add more code...